journalpump/journalpump.py

```diff
@@ -197,23 +198,39 @@ def get_state(self):
             "health": {
                 "status": _convert_to_health(running=self.running, connected=self._connected)[0],
                 "elapsed": time.monotonic() - self._connected_changed,
+                "errors": "\n".join(self._errors)
             }
         }
 
     def mark_connected(self):
         if not self._connected:
             self._connected_changed = time.monotonic()
         self._connected = True
+        self._errors = []
 
-    def mark_disconnected(self):
+    def mark_disconnected(self, error=None):
         if self._connected:
             self._connected_changed = time.monotonic()
         self._connected = False
 
+        if error is None:
+            return
+
+        if isinstance(error, str):
+            msg = error
+        elif isinstance(error, Exception):
+            msg = self._errors.append(getattr(error, 'message', repr(error)))
+        else:
+            msg = repr(error)
+
+        if msg and msg not in self._errors:
+            self._errors.append(msg)
```

Comment:
should have only one append place, and cap length of _errors (I recommend keeping most recent ones but YMMV, first one might be also interesting)